src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1217,6 +1219,54 @@ public override int GetHashCode()
             return (this.Name.GetHashCode());
         }
 
+        internal static unsafe int GetIgnoreCaseHash(string source)
+        {
+            Debug.Assert(source != null, "source must not be null");
+
+            // Do not allocate on the stack if string is empty
+            if (source.Length == 0)
+            {
+                return source.GetHashCode();
+            }
+
+            char[] borrowedArr = null;
+            Span<char> span = source.Length <= 255 ?
+                stackalloc char[255] :
+                (borrowedArr = ArrayPool<char>.Shared.Rent(source.Length));
+
+            char c;
+            // We assume that str contains only ASCII characters until
+            // we hit a non-ASCII character to optimize the common case.
+            for (int i = 0; i < source.Length; i++)
+            {
+                c = source[i];
+
+                if (c >= 0x80)
```

Comment:
>0x80 [](start = 25, length = 4)

yo may check against 0xFFFF if the _invariantMode is true. so we can avoid calling ToUpper again